ping/ping_common.c

```diff
@@ -293,6 +294,48 @@ void print_timestamp(struct ping_rts *rts)
 	}
 }
 
+/* Helper functions for constructing RFC 8335 PROBE messages */
+/* Verifies name is a valid linux interface name
+ * Copied from iproute2's if name check
+ */
+int check_ifname(const char *name)
+{
+	/* These checks mimic kernel checks in dev_valid_name */
+	if (*name == '\0')
+		return -1;
+	if (strlen(name) >= IFNAMSIZ)
+		return -1;
+
+	while (*name) {
+		if (*name == '/' || isspace(*name))
```

Comment:
As mentioned above, slashes are common in interface names.  RFC8335 only limits interface names by referring to RFC7223, which says that they are yang strings.